auth/client/iam/client.go

```diff
@@ -120,3 +128,47 @@ func (hb HTTPClient) PresentationDefinition(ctx context.Context, definitionEndpo
 
 	return &presentationDefinition, nil
 }
+
+func (hb HTTPClient) AccessToken(ctx context.Context, tokenEndpoint string, vp vc.VerifiablePresentation, submission pe.PresentationSubmission, scopes string) (oauth.TokenResponse, error) {
+	var token oauth.TokenResponse
+	presentationDefinitionURL, err := url.Parse(tokenEndpoint)
+	if err != nil {
+		return token, err
+	}
+
+	// create a POST request with x-www-form-urlencoded body
+	assertion, _ := json.Marshal(vp)
+	presentationSubmission, _ := json.Marshal(submission)
+	data := url.Values{}
+	data.Set("grant_type", "vp_token-bearer")
+	data.Set("assertion", string(assertion))
+	data.Set("presentation_submission", string(presentationSubmission))
+	data.Set("scope", scopes)
+	request, err := http.NewRequestWithContext(ctx, http.MethodPost, presentationDefinitionURL.String(), strings.NewReader(data.Encode()))
+	request.Header.Add("Content-Type", "application/x-www-form-urlencoded")
+	if err != nil {
+		return token, err
+	}
+	response, err := hb.httpClient.Do(request.WithContext(ctx))
+	if err != nil {
+		return token, fmt.Errorf("failed to call endpoint: %w", err)
+	}
+	if err = core.TestResponseCode(http.StatusOK, response); err != nil {
+		// check for oauth error
+		if innerErr := core.TestResponseCode(http.StatusBadRequest, response); innerErr == nil {
+			// a non oauth error, the response body could contain a lot of stuff. We'll log and return the entire error
+			log.Logger().Errorf("authorization server token endpoint returned non oauth error (statusCode=%d)", response.StatusCode)
```

Comment:
certainly not on ERROR

Reply:
What do you mean?

Reply:
No need to log it on error, i'd say info or maaaybe warn?

Reply:
I remember why putting it here: to be able to distinguish between remote and local errors. In the end it'll become a 500 at the API level. During development you want some additional hints on what is going on. The entire error will be logged at the API level anyway. Changed to debug